lib/idx/remediate.ts

```diff
@@ -0,0 +1,52 @@
+/* eslint-disable max-statements */
+/* eslint-disable complexity */
+import { IdxResponse, isRawIdxResponse, RemediationValues, Remediator, IdxRemediation } from '../types';
+import { createApiError, isErrorResponse } from './util';
+import Identify from './remediatiors/Identify';
+import ChallengeAuthenticator from './remediatiors/ChallengeAuthenticator';
+
+const REMEDIATORS = {
+  'identify': Identify,
+  'challenge-authenticator': ChallengeAuthenticator
+  // add more
+};
+
+// This function is called recursively until it reaches success or cannot be remediated
+export async function remediate(
+  idxResponse: IdxResponse,
+  values: RemediationValues
+) {
+  const { neededToProceed } = idxResponse;
+  const idxRemediation = neededToProceed[0];
```

Comment:
why only check the first one?

Reply:
My understanding is we can check the supported `REMEDIATORS` from the `neededToProceed` list, then use the first match to proceed.
Here the assumption is based on the assigned policy, only one `proceed` should match (in happy path).

Reply:
OK, that's a good note - we can find the first available remediator that matches.  The assumption is that if we submit values to satisfy a remediation we will get back a DIFFERENT response. I hope this assumption will hold no matter which remediation we choose to proceed with, otherwise we will see infinite loop.

Reply:
Or we can keep previous visits in memory as a loop breaker (just show an error state if the flow breaks unexpectedly).